src/satellite_images_nso/api/nso_georegion.py

```diff
@@ -141,6 +189,22 @@ def retrieve_download_links(
             cloud_coverage_whole,
         )
 
+        if find_nearest_to_previous_link is True:
+
+            # Find the link closed to the previous link.
```

Comment:
Why are we looking for the link closest to the previous link? What not the closest to the current link? What is this previous link supposed to represent?

Reply:
The previous link the link with the missing region this code block tries to find links which are matched in date the closed to the link with the missing region

Reply:
it's a object in a object thus the name previous link, it's a bit of recursion until the whole missing region is filled